lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -654,10 +660,193 @@ namespace Intl
+    int GetPatternForSkeleton(_In_z_ const char *langtag, _In_z_ const char16 *skeleton, _Out_writes_opt_(patternLen) char16 *pattern, _In_ int patternLen)
+    {
+        UErrorCode status = U_ZERO_ERROR;
+        char localeID[ULOC_FULLNAME_CAPACITY] = { 0 };
+        int length = 0;
+        uloc_forLanguageTag(langtag, localeID, ULOC_FULLNAME_CAPACITY, &length, &status);
+        ICU_ASSERT(status, length > 0);
+
+        UDateTimePatternGenerator *dtpg = udatpg_open(localeID, &status);
+        ICU_ASSERT(status, true);
+
+        int bestPatternLen = udatpg_getBestPatternWithOptions(
+            dtpg,
+            reinterpret_cast<const UChar *>(skeleton),
+            -1,
+            UDATPG_MATCH_ALL_FIELDS_LENGTH,
+            reinterpret_cast<UChar *>(pattern),
+            patternLen,
+            &status
+        );
+        ICU_ASSERT(status, bestPatternLen > 0 && bestPatternLen < patternLen);
+
+        udatpg_close(dtpg);
+        return bestPatternLen + 1;
+    }
+
+    // creates a UDateFormat and wraps it in an IPlatformAgnosticResource
+    void CreateDateTimeFormat(_In_z_ const char *langtag, _In_z_ const char16 *timeZone, _In_z_ const char16 *pattern, _Out_ IPlatformAgnosticResource **resource)
+    {
+        UErrorCode status = U_ZERO_ERROR;
+        char localeID[ULOC_FULLNAME_CAPACITY] = { 0 };
+        int length = 0;
+        uloc_forLanguageTag(langtag, localeID, ULOC_FULLNAME_CAPACITY, &length, &status);
+        ICU_ASSERT(status, length > 0);
+
+        UDateFormat *dtf = udat_open(
+            UDAT_PATTERN,
+            UDAT_PATTERN,
+            localeID,
+            reinterpret_cast<const UChar *>(timeZone),
+            -1,
+            reinterpret_cast<const UChar *>(pattern),
+            -1,
+            &status
+        );
+        IPlatformAgnosticResource *formatterResource = new IcuCObject<UDateFormat>(dtf, &udat_close); // TODO(jahorto): fix leak
+        AssertOrFailFast(formatterResource);
+        *resource = formatterResource;
+    }
+
+    // Formats `date` using the UDateFormat wrapped by `resource`. If `formatted` is non-null, the result is written into `formatted`
+    // Returns the number of characters written into `formatted` (including a null terminator) [should always be positive]
+    int FormatDateTime(_In_ IPlatformAgnosticResource *resource, _In_ double date, _Out_writes_opt_(formattedLen) char16 *formatted, _In_ int formattedLen)
+    {
+        UErrorCode status = U_ZERO_ERROR;
+        UDateFormat *dtf = UNWRAP_COBJECT(resource, UDateFormat);
+
+        int required = udat_format(dtf, date, reinterpret_cast<UChar *>(formatted), formattedLen, nullptr, &status);
+        ICU_ASSERT(status, required > 0 && required < formattedLen);
+
+        return required + 1;
+    }
+
+    // Formats `date` using the UDateFormat wrapped by `resource`. If `formatted` is non-null, the result is written into `formatted`
```

Comment:
This looks like a duplicated comment from the above function, that doesn't describe how the `fieldIterator` works

Reply:
Fixed (see line below)